tests/memoryview/memoryview_annotation_typing.py

```diff
@@ -75,7 +75,7 @@ def slice_none(m: cython.double[:]):
     return 1 if m is None else 2
 
 
-def slice_optional(m: typing.Optional[cython.double[:]]):
+def slice_optional(m: cython.double[:] | None):
```

Comment:
This is changing what's being tested (and I don't think the new version is yet understood by Cython's typing system)